specs/predeploys.md

```diff
@@ -0,0 +1,670 @@
+ * contract's `sentMessages` mapping.
+ */
+interface iOVM_L2ToL1MessagePasser {
+    event L2ToL1Message(uint256 _nonce, address _sender, bytes _data);
+
+    /**
+     * Returns a bool if the message has been passed to L1
+     * @param _msgHash Message hash
+     */
+    function sentMessages(bytes32 _msgHash) public returns (bool);
+
+    /**
+     * Passes a message to L1.
+     * @param _message Message to pass to L1.
+     */
+    function passMessageToL1(bytes calldata _message) external;
+}
+```
+
+## OVM\_DeployerWhitelist
+
+The `OVM_DeployerWhitelist` had arbitrary contract deployment enabled. In the
+legacy system, this contract was hooked into during `CREATE` and `CREATE2` to
+ensure that the deployer was allowlisted. In the bedrock system, this will
+be removed from the `CREATE` codepath since arbitrary contract deployment
+has already been enabled.
+
+```solidity
+/**
+ * @title OVM_DeployerWhitelist
+ * @dev The Deployer Whitelist is a temporary predeploy used to provide additional safety during the
+ * initial phases of our mainnet roll out. It is owned by the Optimism team, and defines accounts
+ * which are allowed to deploy contracts on Layer2. The Execution Manager will only allow an
+ * ovmCREATE or ovmCREATE2 operation to proceed if the deployer's address whitelisted.
+ */
+interface iOVM_DeployerWhitelist {
+    event OwnerChanged(address,address);
+    event WhitelistStatusChanged(address,bool);
+    event WhitelistDisabled(address);
+
+    /**
+     * @dev Returns the owner of the contract
+     */
+    function owner() public return (address);
+    /**
+     * @dev Query if an address is in the allowlist
+     */
+    function whitelist(address) public returns (bool);
+
+    /**
+     * @dev Adds or removes an address from the deployment whitelist.
+     * @param _deployer Address to update permissions for.
+     * @param _isWhitelisted Whether or not the address is whitelisted.
+     */
+    function setWhitelistedDeployer(address _deployer, bool _isWhitelisted) external;
+
+    /**
+     * @dev Updates the owner of this contract.
+     * @param _owner Address of the new owner.
+     */
+    function setOwner(address _owner) public;
+
+    /**
+     * @dev Permanently enables arbitrary contract deployment and deletes the owner.
+     */
+    function enableArbitraryContractDeployment() external;
+
+    /**
+     * @dev Checks whether an address is allowed to deploy contracts.
+     * @param _deployer Address to check.
+     * @return _allowed Whether or not the address can deploy contracts.
+     */
+    function isDeployerAllowed(address _deployer) external view returns (bool);
+}
+```
+
+## OVM\_ETH
+
+The `OVM_ETH` contains the ERC20 represented balances of ETH that has been
+deposited to L2. As part of the bedrock upgrade, the balances will be migrated
+from this contract to the actual Ethereum level accounts to preserve EVM
+equivalence.
```

Comment:
What happens to the public functions here, will they still work?

Reply:
I've added a note that this contract's usage should be avoided, some important functions (`transfer`) do not work currently